### DISCUSSION TITLE
MQTT Connection errors on 2.3.7

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#321] [Q&A] [answered]
Good afternoon - installed 2.3.7 to connect to my Home Assistant setup, but for some reason having some connection issues on MQTT.

On my AqualinkD Pi Zero 2W - I have the Aqualinkd.conf file with the MQTT settings (IP address/port pointing to my Home Assistant, username, password, topic, HA autodetect) - all 5 of those lines are active:

On my Home Assistant (RPi4) - I've got the Mosquitto broker running as an Add-on, and it then auto-detects a MQTT Integration that I configure and it then fails because it can't connect. 

When I use the AqualinkD Management Console (Web UI) - I keep getting MQTT errors and connection closes.   Same in my Home Assistant Mosquitto log.

This is from the AqualinkD WebUI:
Warning:NetService:Got mqtt connection error: 5
Notice:NetService:Starting MQTT client to {homeassistant IP address}:1883
Warning:NetService:MQTT Connection closed
Warning:NetService:Got mqtt connection error: 5

This is from Home Assistant Mosquitto broker log:
2024-07-13 16:22:42: New connection from {PiZero IP address}:44000 on port 1883.
2024-07-13 16:22:42: Client aqualinkd_d83addbc disconnected, not authorised.
2024-07-13 16:23:00: New connection from {PiZero IP address}:40582 on port 1883.
2024-07-13 16:23:00: Client aqualinkd_d83addbc disconnected, not authorised.
2024-07-13 16:23:15: New connection from {PiZero IP address}:33910 on port 1883.
2024-07-13 16:23:15: Client aqualinkd_d83addbc disconnected, not authorised.
2024-07-13 16:23:41: New connection from {PiZero IP address}:52496 on port 1883.
2024-07-13 16:23:41: Client aqualinkd_d83addbc disconnected, not authorised.

Any ideas on why these aren't talking?   Appreciate the help... i'm sure i'm doing something dumb that I cannot figure out!

**Comment by sfeakes** [comment, marked as answer]
Setting up Mosquito broker on Home Assistant can be a real PITA.  From the looks of the messages it seems Mosquito is rejecting the user/paswd you have configured in AqualinkD.  This isn't surprising since the HA documentation seems to be currently wrong for how you setup user/passwd in mosquitto. (well it didn't work for me when I setup a dev HA install to develop this integration).

Test this by installing mosquitto_clients on any machine, and use that to test the login, it'll be easier and quicker than constantly editing aqualinkd.conf and restarting. Once you get that working, then configure AqualinkD.

see this post for details.
https://github.com/sfeakes/AqualinkD/discussions/292#discussioncomment-9423259
Once you get mosquitto_sub working then you should be good.

As for HA configuration, I seem to remember I fixed that by setting a user/password in the MQTT Broker configuration and NOT using a system user/passwd.  The HA document suggests you do the latter.
If you are still having problems, ie can't get mosquitto_sub to connect, you should post in the HA forums, you'll have more experts there.

**Reply by jmp2000** [reply]
You're exactly correct (as usual!) - what I needed to do was in the Mosquitto Broker add-on page, once you have it installed - there's a configuration tab.

I added the following to the Config tab under Logins in the MQTT Add-on for Home Assistant:

logins:
  - username: abcdefg
    password: hijklmnop
and there's a ton of other stuff in there by default, i left it all alone.

It's weird, but 2 spaces, and a dash, then a space, then username, colon, then a space, then your username
Then for password, line up right under username, but no dash on that one.

Then you save it, and configure MQTT as an Integration - and then BOOM everything syncs and works.

And then what's even weirder - is when you go back to MQTT - it only shows:
  - username: abcdefg
    password: hijklmnop

But it works!   Thanks again... so strange.... MQTT is a PITA to get setup on Home Assistant!
